tests/test_subroutines.py

```diff
@@ -208,8 +208,11 @@ def test_update_weights_matrix(tuwm):
     actual = update_weights_matrix(
         tuwm[0][0], tuwm[0][1], tuwm[0][2], tuwm[0][3], tuwm[0][4], tuwm[0][5], tuwm[0][6], tuwm[0][7]
     )
+
+    print("Actual test get stretched component", actual)
+    print("Expected test get stretched component", tuwm[1])
     expected = tuwm[1]
-    np.testing.assert_allclose(actual, expected, rtol=1e-03, atol=0.5)
```

Comment:
`atol=0.5` is way too high considering we are comparing numbers between 0 and 1.2, etc.